packages/example-package/src/index.ts

```diff
@@ -25,6 +25,6 @@ export function generateGreeting(): Greeting {
  * A greeting with a message and a proejct name.
  */
 export interface Greeting {
-  message: String
-  project: String
+  message: string
```

Comment:
Never use `String`, always `string`. `String` represents non-primitive object instance that's made of String.prototype. Strings should never be created this way and TS should never check that. 

Let me show you. 

Run-time difference

```
const a = new String("lorem")
const b = new String("lorem")
typeof a // results with object
a === b // results with false, because both are objects with different references
```

Second, TS difference: https://www.typescriptlang.org/play/?#code/JYOwLgpgTgZghgYwgAgCoE8AOKDeBYAKGWOTgC5kBlMKUAc0JOQCMKBnG+wgX0MIQD2IDqQoZsyALzJ8REuRm85xVooA0PIA

See type errors that it produced.

Reply:
Nice; thanks for the context!